### PR TITLE
Exposes some `gm convert` functionalities.

### PR DESCRIPTION
**Pull request by thalesmg**
Exposes `BlackThresholdImage` and `ImageThreshold`.

### CODE REVIEW

File: lib/c/exmagick.c [hunk 1]
```diff
@@ -117,6 +120,17 @@ int exmagick_get_boolean_u (ErlNifEnv *env, ERL_NIF_TERM arg, unsigned int *p)
   return(ecode);
 }
 
+static
+int exmagick_get_double (ErlNifEnv *env, ERL_NIF_TERM arg, double *dbl)
+{
+  int ecode;
+  char atom[EXM_MAX_ATOM_SIZE];
```
Comment (soteras) [inline lib/c/exmagick.c]: warning: unused variable 'atom' [-Wunused-variable]

File: lib/c/exmagick.c [hunk 2]
```diff
@@ -572,3 +586,48 @@ ERL_NIF_TERM exmagick_image_dump_blob (ErlNifEnv *env, int argc, const ERL_NIF_T
...
+  if (strcmp("black_threshold_image", atom) == 0)
+  {
+    char str_value[MaxTextExtent];
+
+    if (0 == exmagick_get_utf8str(env, argv[2], &utf8))
+    { EXM_FAIL(ehandler, "argv[2]: bad argument"); }
+
+    exmagick_utf8strcpy(str_value, &utf8, MaxTextExtent);
+
+    if (0 == BlackThresholdImage(resource->image, &str_value))
```
Comment (soteras) [inline lib/c/exmagick.c]: warning: incompatible pointer types passing 'char (*)[2053]' to parameter of type 'const char *' [-Wincompatible-$ointer-types]

**Comment by soteras** [comment]
You should check that warnings.

graphicsmagick: 1.3.29

**Comment by soteras** [comment]
- Removed warnings.
- New option `WhiteThresholdImage`.
- Update `convert/3` doc.

File: test/exmagick_test.exs [hunk 3]
```diff
@@ -233,6 +233,14 @@ defmodule ExMagickTest do
   end
 
   describe "convert/3" do
+    test "treshold_image", context do
```
Comment (thalesmg) [inline test/exmagick_test.exs]: `threshold`

**Pull request merged**